File: models/folder.go
package models

import (
	"time"
)

type Folder struct {
	ID        uint       `json:"id" gorm:"primaryKey"`
	Name      string     `json:"name" binding:"required"`
	UserID    uint       `json:"user_id"`
	CreatedAt time.Time  `json:"created_at,omitempty"`
	UpdatedAt time.Time  `json:"updated_at,omitempty"`
	DeletedAt *time.Time `json:"deleted_at,omitempty" gorm:"index"`
}

File: mobile_app/lib/models/folder.dart
class Folder {
  final int id;
  final String name;
  final int userId;
  final DateTime createdAt;
  final DateTime updatedAt;

  Folder({
    required this.id,
    required this.name,
    required this.userId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Folder.fromJson(Map<String, dynamic> json) {
    return Folder(
      id: json['id'] is String ? int.parse(json['id']) : json['id'],
      name: json['name'],
      userId: json['user_id'] is String
          ? int.parse(json['user_id'])
          : json['user_id'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'user_id': userId,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}
